drivers/base/core.c

```diff
@@ -4300,7 +4300,7 @@ void set_secondary_fwnode(struct device *dev, struct fwnode_handle *fwnode)
 	if (fwnode)
 		fwnode->secondary = ERR_PTR(-ENODEV);
 
-	if (fwnode_is_primary(dev->fwnode))
+	if (dev->fwnode)
```

Comment:
The change may be right, but I don't see how this fixes the timeouts and what the relationship is between pm_runtime issues and device properties?
Also your commit message is hard to follow, can you revisit the wording and explain the sequence that leads to the problem better?

Reply:
It's also not clear to me why this change is only required *after* @andy-shev changes - Really lost on how Andy's changes introduced the resume timeouts we are seeing and how this patches corrects the behavior.